aws/resource_aws_instance.go

```diff
@@ -1528,6 +1543,46 @@ func readSecurityGroups(d *schema.ResourceData, instance *ec2.Instance, conn *ec
 	return nil
 }
 
+func readPasswordData(d *schema.ResourceData, instance *ec2.Instance, conn *ec2.EC2) error {
+	if !d.Get("get_password_data").(bool) {
+		if err := d.Set("password_data", nil); err != nil {
```

Comment:
When using `d.Set()` with `*string`, we do not need the additional `err` checking. 👍